Standards/scs-0114-v1-secure-connections.md

```diff
@@ -0,0 +1,172 @@
+
+There are many communication channels in OpenStack with different characteristics, location and eligible means of protection.
+Not all channels are equally easy to secure and some protection mechanisms might put unbearable burdens on a CSP.
+Hence, careful assessment is required to determine for which SCS will either mandate or recommend the use of a protection mechanism.
+
+For this distinction to be made, communication channels must be categorized and classified accordingly.
+
+### Communication Channels
+
+The following overview will classify the main communication channels.
+
+| # | Classification | Details | Example solution |
+|---|---|---|---|
+| 1 | Database backend traffic | Replication and sync between database instances | SSL/TLS |
+| 2 | Database frontend traffic | Communication between OpenStack services and databases | SSL/TLS |
+| 3 | Message queue traffic | Message queue communication between OpenStack components as provided by oslo.messaging | SSL/TLS |
+| 4 | External API communication | HTTP traffic to services registered as external endpoints in the Keystone service catalog | SSL/TLS |
+| 5 | Internal API communication | HTTP traffic to services registered as internal endpoints in the Keystone service catalog | SSL/TLS |
+| 6 | Nova VM migration traffic | Nova VM migration data transfer traffic between compute nodes | QEMU-native TLS |
+| 7 | External Neutron network traffic | VM-related traffic between the network/controller nodes and external networks (e.g. internet) established through routed provider networks and floating IPs | VPN |
+| 8 | Internal Neutron network traffic | Traffic within Neutron SDN networks exchanged between internal systems such as network/controller and compute nodes | IPsec |
+
+Notes about the classification categories and implications:
+
+1. Most database clustering solutions (e.g. MariaDB Galera) offer TLS-based encryption of their backend channels. This needs no additional configuration in OpenStack and is a configuration solely concerning the database cluster.
+2. The database frontend interface is the primary connection target for the OpenStack services. OpenStack supports using TLS for database connections.
+3. For the message queue, AMQP-based solutions such as RabbitMQ and QPid do offer TLS natively which is also supported by OpenStack. ZeroMQ does not and requires IPsec or CIPSO instead.
+4. External API endpoints can be protected easily by using a TLS proxy. They can then be registered with their HTTPS endpoint in the Keystone service catalog. The certificates of external APIs usually need to be signed by a well-known CA in order to be accepted by arbitrary external clients.
+5. Internal API endpoints can use the same TLS proxy mechanisms as the internal ones. Optionally, the TLS certificate provider and PKI can differ to the internal ones. It is often sufficient for the CA of internal TLS endpoints to be accepted within the infrastructure and doesn't need to be a common public CA.
```

Comment:
"Internal API endpoints can use the same TLS proxy mechanisms as the internal ones."
Did you mean "Internal API endpoints can use the same TLS proxy mechanisms as the external ones."?
The whole point leaves me a little bit confused, maybe it is just the wording :)

Reply:
Yes, this was a copy/paste error, well spotted! I rephrased it for better clarity.